### PR TITLE
Added demo code for utils/similarity features

### PR DESCRIPTION
**Pull request by benjaminLuoHC**
Changes

* Cleaned the existing `report_demo.ipynb`
* Added `similarity_demo.ipynb` for both the `utils` and `Directory.identify_duplicates()` tools
* Updated imports for `utils` and added code in `__init__.py` to simplify imports
* Added unit tests for cosine similarity and Levenshtein distance

Resolves issue #141

### CODE REVIEW

File: tests/test_file_similarity.py
Comment (pgaviganHC) [inline tests/test_file_similarity.py:1]: @BrennanHCSC: Can you please confirm that the coverage of these test cases is appropriate?

**Comment by benjaminLuoHC** [comment]
I added the `gitignore` change for the CSV file created by `logging_demo.ipynb` here

**Review by pgaviganHC — COMMENTED** [review]
Same issue with the directory test (#144) but all other tests are passing.

Code looks good to me. @BrennanHCSC: Can you please confirm that the distance calculation tests are complete? Once confirmed I believe that this PR can be approved.

**Comment by benjaminLuoHC** [comment]
> Same issue with the directory test (#144) but all other tests are passing.
> 
> Code looks good to me. @BrennanHCSC: Can you please confirm that the distance calculation tests are complete? Once confirmed I believe that this PR can be approved.

Did we ever determine whether it's the code that's causing the issue or the tests? I'm unable to replicate the issue but if it's test-induced then I can write a mock for it as discussed yesterday

**Comment by benjaminLuoHC** [comment]
I see the concern with the tests. The similarity feature itself is very simple and doesn't have exception handlers for non-document-based files:

image

I'll add the check and test right now

**Review by BrennanHCSC — APPROVED** [review]

**Pull request merged**